tests/delete/tests.py

```diff
@@ -349,6 +350,22 @@ def test_large_delete_related(self):
         self.assertFalse(S.objects.exists())
         self.assertFalse(T.objects.exists())
 
+    def test_delete_with_no_parents_deletion(self):
+        child = RChild.objects.create()
+        parent_id = child.r_ptr_id
+        child.delete(keep_parents=True)
+        self.assertFalse(RChild.objects.filter(id=child.id).exists())
+        self.assertTrue(R.objects.filter(id=parent_id).exists())
+
+    def test_delete_grandchild_not_affects_parents(self):
```

Comment:
I think this test is sufficient, no need for the other one. Can you rename it to `test_delete_keep_parents()` though.